### PR TITLE
Added SDK calls in tenantID function

### PR DESCRIPTION
**Pull request by Vidya2606**
# Description

Migrating getTenantId function from using the Azure CLI calls to SDK calls from azure-sdk-for-go.

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a  breaking change which will impact consuming tool(s).

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Tatsinnit — COMMENTED** [review]
💡💡☕️ There are `checksum failures`  https://github.com/Azure/draft/pull/258/checks#step:4:52 

Could try this: (not sure if this is known issue and repo owners have other suggestions et. al.)

* remove go.sum : `rm go.sum`
* regenerate go.sum : `go mod tidy`

**Comment by Vidya2606** [comment]
> 💡💡☕️ There are `checksum failures` https://github.com/Azure/draft/pull/258/checks#step:4:52
> 
> Could try this: (not sure if this is known issue and repo owners have other suggestions et. al.)
> 
> * remove go.sum : `rm go.sum`
> * regenerate go.sum : `go mod tidy`

File: cmd/setup-gh.go [hunk 1]
```diff
@@ -203,6 +203,6 @@ func GetAzSubscriptionId(subIds []string) string {
 }
 
 func init() {
-	rootCmd.AddCommand(newSetUpCmd())
-
+	ctx := context.Background()
```
Comment (OliverMKing) [inline cmd/setup-gh.go:206]: there's another way to get the context that's better than setting it like this. ctx should be passed from the cli framework.

https://github.com/Azure/aks-spin-plugin/blob/7d86429bbf2cd5ce17df57ed676063b5834be96f/cmd/init.go#L20

File: pkg/providers/azure.go [hunk 2]
```diff
@@ -176,26 +179,58 @@ func (sc *SetUpCmd) assignSpRole() error {
...
-	sc.tenantId = tenantId
+	if len(tenants) > 1 {
+		return errors.New("multiple tenants found")
+	}
+	sc.tenantId = fmt.Sprint(&tenants[0])
 
 	return nil
 }
 
+func ListTenants(ctx context.Context) ([]armsubscription.TenantIDDescription, error) {
```
Comment (OliverMKing) [inline pkg/providers/azure.go:201]: should this be exported?

File: pkg/providers/azure.go [hunk 3]
```diff
@@ -176,26 +179,58 @@ func (sc *SetUpCmd) assignSpRole() error {
...
+	log.Debug("listing Azure subscriptions")
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
```
Comment (OliverMKing) [inline pkg/providers/azure.go:204]: we should be passing the cred into this function. that's a lot more flexible.

we are going to want to use the same creds / configuration for when we call the sdk. if it's not in a common place we will need to duplicate logic if/when we add support for different azure endpoints, etc

File: pkg/providers/azure.go (hunk 2 above)
Comment (Vidya2606) [reply pkg/providers/azure.go]: Changed it to lower case!

File: pkg/providers/azure.go [hunk 4]
```diff
@@ -176,26 +179,58 @@ func (sc *SetUpCmd) assignSpRole() error {
...
-	var tenantId string
-	if err := json.Unmarshal(out, &tenantId); err != nil {
-		return err
+	if len(tenants) == 0 {
+		return errors.New("no tenants found")
 	}
-	tenantId = fmt.Sprint(tenantId)
-
-	sc.tenantId = tenantId
+	if len(tenants) > 1 {
```
Comment (OliverMKing) [inline pkg/providers/azure.go:193]: are there any cases where we could get back multiple tenants? I don't think there is

**Review by OliverMKing — APPROVED** [review]

File: pkg/providers/azure.go (hunk 4 above)
Comment (Vidya2606) [reply pkg/providers/azure.go:193]: yeah ideally we won't be having multiple tenants but it's good to check in case!!

**Review by bfoley13 — APPROVED** [review]

File: pkg/providers/azure.go [hunk 5]
```diff
@@ -176,26 +179,58 @@ func (sc *SetUpCmd) assignSpRole() error {
...
+		return errors.New("multiple tenants found")
+	}
+	sc.tenantId = fmt.Sprint(&tenants[0])
```
Comment (davidgamero) [inline pkg/providers/azure.go:197]: tenants is a slice of armsubscription.TenantIDDescription, so &tenants[0] is a pointer to a TenantIDDescription. does calling fmt.Sprint(TenantIDDescription) return the TenantID property?

we likely want something similar to `sc.tenantId = *tenants[0].TenantID` to extract the string

Comment (Vidya2606) [reply pkg/providers/azure.go:197]: Addressed the comment in a new PR https://github.com/Azure/draft/pull/259

**Pull request merged**